zone: change default range size to 2GB with min size of 256MB

ajwerner opened the pull request:
This is an experimental change to increase the default range size.
It will only apply to new cluster. This range size is intentionally
very large in order to stress the impact of the change. The hope is to
validate the change at this size with an intention of shipping with
something like 512MB/64MB.

Fixes #39717.

Release note (general change): New clusters will use an increased default range
size of 2GB and will no longer be merged if larger than 128MB as opposed to the
previous 16MB. This change will not affect pre-existing clusters. Range size
settings are controlled through zone configs.

Commit:
zone: change default range size to 16GB with min size of 2GB

This is an experimental change to increase the default range size.
It will only apply to new cluster. This range size is intentionally
very large in order to stress the impact of the change. The hope is to
validate the change at this size with an intention of shipping with
something like 512MB/64MB.

Release note (general change): New clusters will use an increased default range
size of 2GB and will no longer be merged if larger than 128MB as opposed to the
previous 16MB. This change will not affect pre-existing clusters. Range size
settings are controlled through zone configs.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/44941)
<!-- Reviewable:end -->

Commit:
workload: only disable the merge queue for v19.1 and before in Split

Before 19.2 we needed to disable the merge queue in order to issue splits.
There's no good reason to disable it in later versions. Doing so may be
masking problems

Release note: None

ajwerner commented:
https://teamcity.cockroachdb.com/viewLog.html?buildId=1747118&buildTypeId=Cockroach_UnitTests flaked on https://github.com/cockroachdb/cockroach/issues/41941
